### PR TITLE
Optimize hole filling algorithm

### LINKED ISSUE #113: fill_holes is unusably slow for large images
**Is your feature request related to a problem? Please describe.**
Part of the Mesmer post-processing involves 1) removing very small predicted objects and then 2) filling any holes contained within the remaining predicted objects. However, the `fill_holes` function scales very poorly with image size. For small images (512x512), it's around 30% of the post-processing time, but for large images it quickly approaches 99% of the total time. There are two problems with the function:

1) It contains a for loop over all identified holes, which is slow
2) For each identified hole, it creates a binary mask to manipulate. This mask is the same size as the original image, so as the image size increases, so does the mask size, leading to very poor performance

I've copied the function below. 

```
def fill_holes(label_img, size=10, connectivity=1):
    """Fills holes located completely within a given label with pixels of the same value
    Args:
        label_img: a 2D labeled image
        size: maximum size for a hole to be filled in
        connectivity: the connectivity used to define the hole
    Returns:
        label_img_filled: a labeled image with small holes filled in with same value as the label
    """

    label_img_filled = copy(label_img)

    # invert the image, so that holes are objects
    inverse_img = np.logical_not(label_img)

    inverse_img_removed = remove_small_objects(ar=inverse_img, min_size=size,
                                               connectivity=connectivity)

    # identify potential holes in the image
    potential_holes = inverse_img != inverse_img_removed
    potential_holes = label(potential_holes)

    # for each identified hole, check if it is contained completely within a single label
    for i in np.unique(potential_holes):
        mask = potential_holes == i
        enlarged_mask = dilation(mask, selem=square(3))

        label_values = np.unique(label_img[enlarged_mask])
        label_values = [x for x in label_values if x != 0]

        # the enlarged mask only contains a single value, hence the hole is completely contained
        # within a single label
        if len(label_values) == 1:
            label_img_filled[mask] = label_values[0]

    return label_img_filled
```

**Describe the solution you'd like**
I see two potential solutions. The first would be to check the input image size to application, and if it's above a certain threshold, raise a warning to the user that they should turn off file holes. @oldridge, you can manually override this right now when you call predict to set fill_holes=0, you should be able to process your large images. This would be easy to do, but would result in slightly worse segmentation predictions. 

The second would be to optimize this function to work better. We could use coordinate indexing from regionprops instead of mask indexing for a massive speedup. However, I'm not sure if there's a good way to substitute the mask in the "enlarge" step? Any suggestions here would be welcome.

### PR DESCRIPTION
**Pull request by ngreenwald**
This PR closes #113. It replaces the previous hole-filling algorithm with an optimized, regionprops-based version. The remaining bottleneck in the post-processing is now peak finding, so the warning message for large images is now raised when `h_maxima` is selected for large images instead of `peak_local_max`

### CODE REVIEW

**Comment by ngreenwald** [comment]
Your version is only slightly slower than Dave's but handles the corner cases, I don't think we need to have two options anymore since this is now only a small proportion of the processing time

**Review by willgraf — CHANGES_REQUESTED** [review]
I just commented in the issue as well, but I've updated the regionprops build a bit by using more intelligent properties. Can you please update this PR and re-request review?

File: deepcell_toolbox/utils.py [hunk 1]
```diff
@@ -37,9 +37,10 @@
 from scipy.ndimage import fourier_shift
 from skimage.feature import register_translation
 from skimage import transform
-from skimage.segmentation import find_boundaries
-from skimage.morphology import remove_small_objects, square, dilation
-from skimage.measure import label
+from skimage.segmentation import find_boundaries, watershed
+from skimage.morphology import remove_small_objects, remove_small_holes
+from skimage.morphology import square, dilation
+from skimage.measure import label, regionprops
```
Comment (willgraf) [inline deepcell_toolbox/utils.py:43]: Remove the unused imports

```suggestion
from skimage.measure import regionprops
from skimage.morphology import remove_small_holes
from skimage.segmentation import find_boundaries
```

**Review by willgraf — APPROVED** [review]

**Comment by oldridge** [comment]
Thank you everyone for following up on this. I've been enjoying following the discussion, and I greatly appreciate your professionalism and efficiency in dealing with this issue. I can confirm this fix on my end as well, the new code works great!

**Comment by ngreenwald** [comment]
Awesome, glad to hear it

**Pull request merged**